tensorrt_llm/_torch/auto_deploy/config/default.yaml

```diff
@@ -92,13 +93,16 @@ transforms:
   ############################################################################################
   fuse_gemms:
     stage: post_load_fusion
-    enabled: false # TODO: https://github.com/NVIDIA/TensorRT-LLM/issues/4674 this is causing OOMs
+    enabled: true # TODO: https://github.com/NVIDIA/TensorRT-LLM/issues/4674 this is causing OOMs
+    run_shape_prop: true
```

Comment:
`run_shape_prop` is expensive. Let's be a little more rigorous when we apply it.

My suggestion: set `requires_shape_prop: true` for `fuse_allreduce_residual_rmsnorm`. Then it's run as pre-cleanup steam for `fuse_allreduce_residual_rmsnorm` instead of as post-clean up step for every fusion before